### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by Mahamadou-Nouridine**
In this Pr, my coding partner and I brought changes that include:

- Add unit tests for all the methods in all the classes:
  - `Book`
  - `Person`
  - `Student`
  - `Teacher`
  - `Classroom`
  - `Rental`
  - all classes that you created while implementing the Decorator design pattern

### CODE REVIEW

File: Rspec/spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        'title' => title,
+        'author' => author
+      }.to_json
+      new_book = Book.from_json(json_data)
+      expect(new_book).to be_an_instance_of(Book)
+      expect(new_book.title).to eq(title)
+      expect(new_book.author).to eq(author)
+    end
+  end
+end
```
Comment (ghost) [inline Rspec/spec/book_spec.rb]: - [ ] Kindly add a test for the `add_rental` method in your books class to ensure all methods are tested in a class

File: Rspec/spec/person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        'type' => 'Person',
+        'id' => person.id,
+        'age' => age,
+        'name' => name,
+        'parent_permission' => parent_permission
+      }.to_json
+      expect(person.to_json).to eq(expected_json)
+    end
+  end
+end
```
Comment (ghost) [inline Rspec/spec/person_spec.rb]: - [ ] The same case in your person class. Add a test for `add_rental` method to ensure all methods are tested in a class

**Review by frank1738 — CHANGES_REQUESTED** [review]
Hi Team , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

✅ Correct use of Gitflow
✅ No linter error

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by binhussen — APPROVED** [review]
Hi @Mahamadou-Nouridine and @Kel-nana  , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Added unit tests✔️
- Well Structured Code ✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**